### PR TITLE
Update links to mailing lists and Slack

### PR DESCRIPTION
**Pull request by scholzj**
This PR upúdates the mailing list and Slack links to the new CNCF hosted links.

### CODE REVIEW

File: contributing/index.md [hunk 1]
```diff
@@ -20,5 +20,5 @@ The [Documentation Contributor Guide](https://strimzi.io/contributing/guide/) de
 
 If you want to get in touch with us first before contributing, you can use:
 
-* [Strimzi mailing list](https://www.redhat.com/mailman/listinfo/strimzi)
-* [Strimzi Slack workspace](https://join.slack.com/t/strimzi/shared_invite/enQtMzU2Mjk3NTgxMzE5LTYyMTUwMGNlMDQwMzBhOGI4YmY4MjhiMDgyNjA5OTk2MTFiYjc4M2Q3NGU1YTFjOWRiMzM2NGMwNDUwMjBlNDY)
+* [#strimzi channel on CNCF Slack](https://slack.cncf.io/)
```
Comment (ppatierno) [inline contributing/index.md]: This is the link that CNCF seems to use in their sendbox projects page:

https://cloud-native.slack.com/messages/strimzi

File: index.md [hunk 2]
```diff
@@ -42,8 +42,8 @@ Documentation for all releases can be found in the [Documentation](/documentatio
 
 If you encounter any issues while using Strimzi, you can get help using:
 
-* [Strimzi mailing list](https://www.redhat.com/mailman/listinfo/strimzi)
-* [Strimzi Slack workspace](https://join.slack.com/t/strimzi/shared_invite/enQtMzU2Mjk3NTgxMzE5LTYyMTUwMGNlMDQwMzBhOGI4YmY4MjhiMDgyNjA5OTk2MTFiYjc4M2Q3NGU1YTFjOWRiMzM2NGMwNDUwMjBlNDY)
+* [#strimzi channel on CNCF Slack](https://slack.cncf.io/)
```
Comment (ppatierno) [inline index.md]: ditto as above.

File: contributing/index.md (hunk 1 above)
Comment (scholzj) [reply contributing/index.md]: That links is useless unless you are already registered. So I think we should keep what I have there.

File: index.md (hunk 2 above)
Comment (scholzj) [reply index.md]: That links is useless unless you are already registered. So I think we should keep what I have there.

**Pull request merged**